src/main/java/frc/robot/subsystems/elevator/ElevatorSubsystem.java

```diff
@@ -69,14 +73,27 @@ public ElevatorSubsystem() {
 
     @Override 
     public void periodic() {
+        CommandScheduler.getInstance().run();
  
         if (isManual) {
             //Add some factors for better control.
             extensionMotor.set(this.manualPower);
         }
          
-        if (zeroLimitSwitch != null && !zeroLimitSwitch.get()) {
+        if (zeroLimitSwitch != null && !zeroLimitSwitch.get() && ElevatorConstants.LIMIT_SWITCH_ENABLED) {
+            //if limit switch tells us it's at the bottom
             extensionEncoder.setPosition(0); 
+            this.setManualPower(0);//stops the motor in manual mode to avoid motor stall.
+        }
+
+        if (ElevatorConstants.LIMIT_SWITCH_ENABLED) {
+            if (zeroLimitSwitch != null && zeroLimitSwitch.get()) { //if limit switch is not pressed
+                if (Math.abs(this.getExtensionPercent()) < ElevatorConstants.EXTENSION_TOLERANCE) { 
+                    //if the encoder thinks the elevator is at ground
+                    CommandScheduler.getInstance().schedule(new ElevatorToLimitSwitchCommand(this));
```

Comment:
I don't think this will work as you expect. Image we are trying to go to the amp height and it begins raising. If it is still within the tolerance this will override the raising and cancel going to the amp.

Reply:
Also this is probably unnecessary. We only really need to check that it hit the limit switch when we are actively lowering it. It won't raise on its own.

Reply:
Ok I'll delete this. So we only go to limit switch when we hit the button for toGround right?